buffer/buffer.go

```diff
@@ -404,6 +407,167 @@ func (b *Buffer) proto_equals(call goja.FunctionCall) goja.Value {
+	})
+}
+
+// readFloatBE reads a big-endian 32-bit floating-point number from the buffer
+func (b *Buffer) readFloatBE(call goja.FunctionCall) goja.Value {
+	return b.readFixedLengthValue(call, 4, func(bb []byte) any {
+		value := binary.BigEndian.Uint32(bb)
+		return math.Float32frombits(value)
+	})
+}
+
+// readFloatLE reads a little-endian 32-bit floating-point number from the buffer
+func (b *Buffer) readFloatLE(call goja.FunctionCall) goja.Value {
+	return b.readFixedLengthValue(call, 4, func(bb []byte) any {
+		value := binary.LittleEndian.Uint32(bb)
+		return math.Float32frombits(value)
+	})
+}
+
+// readInt8 reads an 8-bit signed integer from the buffer
+func (b *Buffer) readInt8(call goja.FunctionCall) goja.Value {
+	return b.readFixedLengthValue(call, 1, func(bb []byte) any {
+		return int8(bb[0])
+	})
+}
+
+// readInt16BE reads a big-endian 16-bit signed integer from the buffer
+func (b *Buffer) readInt16BE(call goja.FunctionCall) goja.Value {
+	return b.readFixedLengthValue(call, 2, func(bb []byte) any {
+		return int16(binary.BigEndian.Uint16(bb))
+	})
+}
+
+// readInt16LE reads a little-endian 16-bit signed integer from the buffer
+func (b *Buffer) readInt16LE(call goja.FunctionCall) goja.Value {
+	return b.readFixedLengthValue(call, 2, func(bb []byte) any {
+		return int16(binary.LittleEndian.Uint16(bb))
+	})
+}
+
+// readInt32BE reads a big-endian 32-bit signed integer from the buffer
+func (b *Buffer) readInt32BE(call goja.FunctionCall) goja.Value {
+	return b.readFixedLengthValue(call, 4, func(bb []byte) any {
+		return binary.BigEndian.Uint32(bb)
+	})
+}
+
+// readInt32LE reads a little-endian 32-bit signed integer from the buffer
+func (b *Buffer) readInt32LE(call goja.FunctionCall) goja.Value {
+	return b.readFixedLengthValue(call, 4, func(bb []byte) any {
+		return binary.LittleEndian.Uint32(bb)
+	})
+}
+
+// readUInt8 reads an 8-bit unsigned integer from the buffer
+func (b *Buffer) readUInt8(call goja.FunctionCall) goja.Value {
+	return b.readFixedLengthValue(call, 1, func(bb []byte) any {
+		return bb[0]
+	})
+}
+
+// readUInt16BE reads a big-endian 16-bit unsigned integer from the buffer
+func (b *Buffer) readUInt16BE(call goja.FunctionCall) goja.Value {
+	return b.readFixedLengthValue(call, 2, func(bb []byte) any {
+		return binary.BigEndian.Uint16(bb)
+	})
+}
+
+// readUInt16LE reads a little-endian 16-bit unsigned integer from the buffer
+func (b *Buffer) readUInt16LE(call goja.FunctionCall) goja.Value {
+	return b.readFixedLengthValue(call, 2, func(bb []byte) any {
+		return binary.LittleEndian.Uint16(bb)
+	})
+}
+
+// readUInt32BE reads a big-endian 32-bit unsigned integer from the buffer
+func (b *Buffer) readUInt32BE(call goja.FunctionCall) goja.Value {
+	return b.readFixedLengthValue(call, 4, func(bb []byte) any {
+		return binary.BigEndian.Uint32(bb)
+	})
+}
+
+// readUInt32LE reads a little-endian 32-bit unsigned integer from the buffer
+func (b *Buffer) readUInt32LE(call goja.FunctionCall) goja.Value {
```

Comment:
Thanks for the PR!

Have you considered a more straightforward approach, i.e.
```go
func (b *Buffer) readUInt32LE(call goja.FunctionCall) goja.Value {
	bb := Bytes(b.r, call.This)
	offset := b.getOffsetArgument(call, 0, bb, 4)
	value := binary.LittleEndian.Uint32(bb[offset : offset+4])

	return b.r.ToValue(value)
}
```

It doesn't make it significantly more verbose, but results in a better assembly being generated and a slightly better performance as a result.

Reply:
sure, I actually had that originally and refactored so I was repeating myself, but I can definitely change it. Oh, also, I typically use `testify` for unit test assertions. Would you be opposed if I brought that in?

Reply:
I'd like to keep the number of dependencies as low as possible.

Reply:
OK, totally understand. Pushed the change to inline that logic.